base_layer/wallet/src/output_manager_service/storage/sqlite_db/mod.rs

```diff
@@ -569,30 +569,54 @@ impl OutputManagerBackend for OutputManagerSqliteDatabase {
         Ok(())
     }
 
-    fn update_last_validation_timestamps(&self, hashes: Vec<FixedHash>) -> Result<(), OutputManagerStorageError> {
+    fn update_last_validation_timestamps(&self, commitments: Vec<Commitment>) -> Result<(), OutputManagerStorageError> {
         let start = Instant::now();
         let mut conn = self.database_connection.get_pooled_connection()?;
         let acquire_lock = start.elapsed();
 
-        // Using a raw query here, as the obvious diesel query is not as performant as expected:
-        // `diesel::update(outputs::table.filter(outputs::hash.eq_any(hashes.iter().map(|hash| hash.to_vec()))))
-        // `    .set(outputs::last_validation_timestamp.eq(Some(Utc::now().naive_utc())))
-        // `     .execute(&mut conn)
-        // `     .num_rows_affected_or_not_found(hashes.len())?;
-        let sql_query = format!(
-            r#"
-            UPDATE outputs
-            SET last_validation_timestamp = '{}'
-            WHERE hash IN ({})
-            "#,
-            Utc::now().naive_utc(),
-            hashes
+        let last_validation_timestamp = Utc::now().naive_utc();
+
+        // Three queries were evaluated to determine the most efficient way to update the last validation timestamp
+        // during system-level stress testing:
+        // - Using `diesel`:
+        //   - `diesel::update(outputs::table.filter(outputs::hash.eq_any(hashes)).set(...).execute(&mut conn)`
+        //   - Note: `diesel` does not support batch updates, so we have to do it manually.
+        // - Using a raw query that mimicked the `diesel` query:
+        //   - `UPDATE outputs SET last_validation_timestamp = '{}' WHERE hash IN ({})`
+        //   - 20% faster than `diesel` on average
+        // - Using a raw query with a batch insert (as implemented below):
+        //   - `INSERT INTO outputs (..) VALUES (...) ON CONFLICT (commitment) DO UPDATE SET ...`
+        //   - 1011% faster than `diesel` on average
+
+        let mut query = String::from(
```

Comment:
This is why I dont like this method that much, is super complicated, you need to add a check here to ensure we don't add rubbish into the table.

Reply:
Yip, I overlooked that pattern